server/datastore/mysql/software.go

```diff
@@ -450,6 +472,10 @@ func (d *Datastore) ListSoftware(ctx context.Context, opt fleet.SoftwareListOpti
 	return listSoftwareDB(ctx, d.reader, nil, opt)
 }
 
+func (d *Datastore) CountSoftware(ctx context.Context, opt fleet.SoftwareListOptions) (int, error) {
```

Comment:
Maybe document here that CountSoftware ignores some fields of `opt`. (I know it's already documented on the API, but it's worth mentioning again here for future readers.)